### PR TITLE
Document src/wp-admin/js/inline-edit-tax.js

### LINKED ISSUE #5: Document wp-admin/js/inline-edit-tax.js

### PR DESCRIPTION
**Pull request by atimmer**
Fixes #5

### CODE REVIEW

File: src/wp-admin/js/inline-edit-tax.js [hunk 1]
```diff
@@ -137,27 +197,47 @@ inlineEditTax = {
...
 	getId : function(o) {
 		var id = o.tagName === 'TR' ? o.id : $(o).parents('tr').attr('id'), parts = id.split('-');
+
 		return parts[parts.length - 1];
 	}
 };
 
-$(document).ready(function(){inlineEditTax.init();});
+$(document).ready(function(){
+	inlineEditTax.init();
```
Comment (jdevalk) [inline src/wp-admin/js/inline-edit-tax.js]: As trivial as this change is, it's what'd be considered a refactor, so don't do this in a docs change.

File: src/wp-admin/js/inline-edit-tax.js [hunk 2]
```diff
@@ -16,8 +32,13 @@ inlineEditTax = {
 			return false;
 		});
 
-		// prepare the edit row
+		/*
+		 * Prepare the edit row.
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: What happens here, binds the escape button to revert the inline editor?

File: src/wp-admin/js/inline-edit-tax.js [hunk 3]
```diff
@@ -1,10 +1,26 @@
 /* global inlineEditL10n, ajaxurl */
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: Missing file headers: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#file-headers

File: src/wp-admin/js/inline-edit-tax.js [hunk 4]
```diff
@@ -1,10 +1,26 @@
...
 ( function( $, wp ) {
+
 inlineEditTax = {
 
+	/**
+	 * Initializes the inline taxonomy editor, adds event handlers to be able to quick edit.
+	 *
+	 * @this inlineEditTax
+	 * @memberof inlineEditTax
+	 */
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: missing access description: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#object-properties

File: src/wp-admin/js/inline-edit-tax.js [hunk 5]
```diff
@@ -16,8 +32,13 @@ inlineEditTax = {
 			return false;
 		});
 
-		// prepare the edit row
+		/*
+		 * Prepare the edit row.
+		 *
+		 * @param {Object} e The keyup event that has been triggered.
+		 */
 		row.keyup( function( e ) {
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: Maybe add some more description to the event binders, for example @listens or @fires: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#functions

File: src/wp-admin/js/inline-edit-tax.js [hunk 6]
```diff
@@ -40,16 +64,36 @@ inlineEditTax = {
...
 	},
 
+	/**
+	 * Toggles the quick edit based on if it is currently shown or hidden.
+	 *
+	 * @this inlineEditTax
+	 * @memberof inlineEditTax
+	 *
+	 * @param {HTMLElement} el An element within the table row or the table row itself that we want to quick edit.
+	 */
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: Missing @return and @access.

File: src/wp-admin/js/inline-edit-tax.js [hunk 7]
```diff
@@ -40,16 +64,36 @@ inlineEditTax = {
...
+	/**
+	 * Shows the quick editor
+	 *
+	 * @this inlineEditTax
+	 * @memberof inlineEditTax
+	 *
+	 * @param {string|HTMLElement} id The ID of the term we want to quick edit or an element within the table row or the
+	 * table row itself.
+	 * @returns {boolean} Always returns false.
+	 */
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: Missing @access

File: src/wp-admin/js/inline-edit-tax.js [hunk 8]
```diff
@@ -75,9 +119,20 @@ inlineEditTax = {
...
+	/**
+	 * Saves the quick edit data to the server and replaces the table row with the HTML retrieved from the server.
+	 *
+	 * @this inlineEditTax
+	 * @memberof inlineEditTax
+	 *
+	 * @param {string|HTMLElement} id The ID of the term we want to quick edit or an element within the table row or the
+	 * table row itself.
+	 * @returns {boolean} Always returns false.
+	 */
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: Missing @access

File: src/wp-admin/js/inline-edit-tax.js [hunk 9]
```diff
@@ -137,27 +197,47 @@ inlineEditTax = {
...
 		// Prevent submitting the form when pressing Enter on a focused field.
 		return false;
 	},
 
+	/**
+	 * Closes the quick edit form.
+	 *
+	 * @this inlineEditTax
+	 * @memberof inlineEditTax
+	 */
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: Missing @return and @access

File: src/wp-admin/js/inline-edit-tax.js [hunk 10]
```diff
@@ -137,27 +197,47 @@ inlineEditTax = {
...
 	},
 
+	/**
+	 * Retrieves the ID of the term of the element inside the table row.
+	 *
+	 * @memberof inlineEditTax
+	 *
+	 * @param {HTMLElement} o An element within the table row or the table row itself.
+	 * @returns {string} The ID of the term based on the element.
+	 */
```
Comment (rensw90) [inline src/wp-admin/js/inline-edit-tax.js]: Missing @access

File: src/wp-admin/js/inline-edit-tax.js (hunk 4 above)
Comment (atimmer) [inline src/wp-admin/js/inline-edit-tax.js]: This is only required for properties and private functions/methods.

**Comment by atimmer** [comment]
https://core.trac.wordpress.org/ticket/37571

**Pull request closed without merging**